geomesa-feature/src/main/scala/org/locationtech/geomesa/feature/FeatureSpecificReader.scala

```diff
@@ -113,6 +114,23 @@ class FeatureSpecificReader(oldType: SimpleFeatureType, newType: SimpleFeatureTy
     sf
   }
 
+  def readWithUserData(reuse: AvroSimpleFeature, in: Decoder): AvroSimpleFeature = {
+    val sf = defaultRead(reuse, in)
+
+    val ar = AvroSimpleFeatureDecodingsCache.getAbstractReader
+
+    val userData = ar.readGenericMap(in)
+    sf.getUserData.clear()
```

Comment:
shouldn't need to clear the user data here, since it will be a newly-created avroSimpleFeature